src/Generator.h

```diff
@@ -328,15 +328,69 @@ class GeneratorParamBase {
 
     const std::string name;
 
+    // overload the set() function to call the right virtual method based on type.
+
+// It's always a bit iffy to use macros for this, but IMHO it clarifies the situation here.
+#define HALIDE_GENERATOR_PARAM_TYPED_SETTER(TYPE) \
+    HALIDE_ALWAYS_INLINE void set(const TYPE &new_value) { set_from_##TYPE(new_value); }
+
+    HALIDE_GENERATOR_PARAM_TYPED_SETTER(bool)
```

Comment:
Fair enough. I don't know of any good workarounds for that off the top of my head.